source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -0,0 +1,169 @@
+#include "extensions/filters/http/aws_lambda/aws_lambda_filter.h"
+
+#include <string>
+#include <vector>
+
+#include "envoy/upstream/upstream.h"
+
+#include "common/common/fmt.h"
+#include "common/common/hex.h"
+#include "common/crypto/utility.h"
+#include "common/http/headers.h"
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/http/well_known_names.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AwsLambdaFilter {
+
+namespace {
+
+constexpr auto filter_metadata_key = "com.amazonaws.lambda";
+constexpr auto egress_gateway_metadata_key = "egress_gateway";
+
+void setHeaders(Http::HeaderMap& headers, absl::string_view function_name) {
+  headers.setMethod(Http::Headers::get().MethodValues.Post);
+  headers.setPath(fmt::format("/2015-03-31/functions/{}/invocations", function_name));
+  headers.setCopy(Http::LowerCaseString{"x-amz-invocation-type"}, "RequestResponse");
+}
+
+/**
+ * Determines if the target cluster has the AWS Lambda metadata on it.
+ */
+bool isTargetClusterLambdaGateway(Upstream::ClusterInfo const& cluster_info) {
+  using ProtobufWkt::Value;
+  const auto& filter_metadata_map = cluster_info.metadata().filter_metadata();
+  auto metadata_it = filter_metadata_map.find(filter_metadata_key);
+  if (metadata_it == filter_metadata_map.end()) {
+    return false;
+  }
+
+  auto egress_gateway_it = metadata_it->second.fields().find(egress_gateway_metadata_key);
+  if (egress_gateway_it == metadata_it->second.fields().end()) {
+    return false;
+  }
+
+  if (egress_gateway_it->second.kind_case() != Value::KindCase::kBoolValue) {
+    return false;
+  }
+
+  return egress_gateway_it->second.bool_value();
+}
+
+} // namespace
+
+FilterSettings::FilterSettings(const std::string& arn, bool payload_passthrough)
```

Comment:
why not inline these constructors in the header?

Reply:
I think this is a refactoring left-over. I'll move it to the header. Thanks!